benchmark/Cajita_SparsePartitionerPerformance.cpp

```diff
@@ -0,0 +1,481 @@
+        std::stringstream local_workload_name;
+        local_workload_name << test_prefix << "compute_local_workload_"
+                            << "domain_size(cell)_" << num_cells_per_dim[c];
+        Cabana::Benchmark::Timer local_workload_timer(
+            local_workload_name.str(), occupy_fraction_size );
+
+        std::stringstream prefix_sum_name;
+        prefix_sum_name << test_prefix << "compute_prefix_sum_"
+                        << "domain_size(cell)_" << num_cells_per_dim[c];
+        Cabana::Benchmark::Timer prefix_sum_timer( prefix_sum_name.str(),
+                                                   occupy_fraction_size );
+
+        std::stringstream total_optimize_name;
+        total_optimize_name << test_prefix << "total_optimize_"
+                            << "domain_size(cell)_" << num_cells_per_dim[c];
+        Cabana::Benchmark::Timer total_optimize_timer(
+            total_optimize_name.str(), occupy_fraction_size );
+
+        std::stringstream single_optimize_name;
+        single_optimize_name << test_prefix << "single_step_optimize_"
+                             << "domain_size(cell)_" << num_cells_per_dim[c];
+        Cabana::Benchmark::Timer single_optimize_timer(
+            single_optimize_name.str(), occupy_fraction_size );
+
+        // loop over all the occupy_fractions
+        for ( int frac = 0; frac < occupy_fraction_size; ++frac )
+        {
+            // compute the number of distinct tiles that will be inserted to the
+            // sparse map
+            int num_insert =
+                static_cast<int>( occupy_fraction[frac] * num_tiles_per_dim *
+                                  num_tiles_per_dim * num_tiles_per_dim );
+
+            // register selected tiles to the sparseMap
+            Kokkos::parallel_for(
+                "insert_tile_to_sparse_map",
+                Kokkos::RangePolicy<exec_space>( 0, num_insert ),
+                KOKKOS_LAMBDA( const int id ) {
+                    sis.insertTile( tiles_view( id, 0 ), tiles_view( id, 1 ),
+                                    tiles_view( id, 2 ) );
+                } );
+
+            for ( int t = 0; t < num_run; ++t )
+            {
+                // ensure every optimization process starts from the same status
+                partitioner.initializeRecPartition(
+                    ave_partition[0], ave_partition[1], ave_partition[2] );
+
+                // compute local workload
+                local_workload_timer.start( frac );
+                partitioner.computeLocalWorkLoad( sis );
+                local_workload_timer.stop( frac );
+
+                // compute prefix sum matrix
+                prefix_sum_timer.start( frac );
+                partitioner.computeFullPrefixSum( comm );
+                prefix_sum_timer.stop( frac );
+
+                // optimization
+                bool is_changed = false;
+                // full timer
+                total_optimize_timer.start( frac );
```

Comment:
I don't think we get much out of this timer - just one optimization timer is enough for now and if we have a strong case to measure one iteration later on we can add it

Reply:
Actually, let's remove the `single_optimize_timer`